web_src/js/modules/dirauto.js

```diff
@@ -19,7 +19,7 @@ export function initDirAuto() {
       for (let i = 0; i < len; i++) {
         const addedNode = mutation.addedNodes[i];
         if (addedNode.nodeType !== Node.ELEMENT_NODE && addedNode.nodeType !== Node.DOCUMENT_FRAGMENT_NODE) continue;
```

Comment:
I think you can remove this line now. It's primary reason was IIRC, to exclude text nodes but with the `nodeName` check below, this is obsolete.

https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

Reply:
No, it can't be removed. It is for the `addedNode.querySelectorAll` below.

Actually, I ever used before:

```js
if (!addedNode.querySelectorAll) continue;
addedNode.querySelectorAll(....)
```

I think it is clear enough and performant.

While you suggested to use the `nodeType` check:

* https://github.com/go-gitea/gitea/pull/23574#pullrequestreview-1347559564

And then 

* https://github.com/go-gitea/gitea/pull/23988/commits/1b34f56566b65ca0557aa3c358d1aa12527c0e88  in #23988

Reply:
Ah, right, I missed the fact that the check below has no `return`.